packages/x-components/src/components/result/base-result-link.vue

```diff
@@ -65,16 +66,32 @@
         []
       );
 
+      /**
+       * The metadata to be injected in the events emitted by the component. This component can emit
+       * have extra events so this record pairs each event to its metadata.
+       */
+      const resultLinkMetadataPerEvent = inject<
+        Partial<
+          Record<
+            PropsWithType<XEventsTypes, Result>,
+            Omit<WireMetadata, 'moduleName' | 'origin' | 'location'>
+          >
+        >
+      >('resultLinkMetadataPerEvent', {});
```

Comment:
What do you think about adding this injection key to the `injection.consts.ts`?
That way we wouldn't repeat this type when we provide the events.

Reply:
I think those only work for the `XInject` and we're not using them anymore in `2.7` so I don't know if we can use those keys in the native `inject`.